diff --git a/src/uPlot.js b/src/uPlot.js
index 0a208f2..7837837 100644
--- a/src/uPlot.js
+++ b/src/uPlot.js
@@ -554,7 +554,7 @@ export default function uPlot(opts, data, then) {
 		legendCols = multiValLegend ? getMultiVals(self, 1, 0) : {_: 0};
 
 		for (let k in legendCols)
-			NULL_LEGEND_VALUES[k] = "--";
+			NULL_LEGEND_VALUES[k] = "—";
 	}
 
 	if (showLegend) {
@@ -641,7 +641,7 @@ export default function uPlot(opts, data, then) {
 
 		for (var key in legendCols) {
 			let v = placeTag("td", LEGEND_VALUE, row);
-			v.textContent = "--";
+			v.textContent = "—";
 			cells.push(v);
 		}
 
@@ -2218,6 +2218,12 @@ export default function uPlot(opts, data, then) {
 				let isFocused = allFocused || i2 == 0 || i2 == i;
 				s._focus = allFocused ? null : isFocused;
 				_setAlpha && setAlpha(i2, isFocused ? 1 : focus.alpha);
+				if (showLegend && legendRows[i2]) {
+                                       if (allFocused || i2 === i)
+                                               legendRows[i2].classList.add("plot-legend-focus");
+                                       else
+                                               legendRows[i2].classList.remove("plot-legend-focus");
+                                }
 			});
 
 			focusedSeries = i;
@@ -2838,8 +2838,10 @@ function uPlot(opts, data, then) {
 			let head = placeTag("tr", LEGEND_THEAD, legendEl);
 			placeTag("th", null, head);

-			for (var key in legendCols)
+			for (var key in legendCols) {
 				placeTag("th", LEGEND_LABEL, head).textContent = key;
+				placeTag("th", LEGEND_LABEL, head).setAttribute("title", key);
+			}
 		}
 		else {
 			addClass(legendEl, LEGEND_INLINE);
@@ -2880,6 +2882,7 @@ function uPlot(opts, data, then) {

 		let text = placeDiv(LEGEND_LABEL, label);
 		text.textContent = s.label;
+		text.setAttribute("title", s.label);

 		if (i > 0) {
 			if (!markers.show)
